src/cache/inmemory/__tests__/writeToStore.ts

```diff
@@ -3997,20 +3997,26 @@ describe("writing to the store", () => {
         }
       }
     `;
-    cache.write({
-      query: injectingQuery,
-      variables: { id: 5 },
-      dataId: "ROOT_QUERY",
-      result: {
-        user: {
-          __typename: "Post",
-          id: "1",
-          title: "Incorrect!",
-          ignore: "User",
-          ignore2: "5",
+    expect(() =>
+      cache.write({
+        query: injectingQuery,
+        variables: { id: 5 },
+        dataId: "ROOT_QUERY",
+        result: {
+          user: {
+            __typename: "Post",
+            id: "1",
+            title: "Incorrect!",
+            ignore: "User",
+            ignore2: "5",
+          },
         },
-      },
-    });
+      })
+    ).toThrow(
+      new InvariantError(
+        '`__typename` is a forbidden field alias name in the selection set for field `user.__typename` in query "<undefined>".'
```

Comment:
```suggestion
        '`__typename` is a forbidden field alias name in the selection set for field `user.__typename` in query "(anonymous)".'
```

We tend to use `(anonymous)` in other error message for queries without names ([example](https://github.com/apollographql/apollo-client/blob/71875402851be42244f1c018b3dabc8a14f892ab/src/core/QueryManager.ts#L293)). Can we do that here? We have that `getOperationName` helper which takes a `default` argument which makes this easy 🙂